security/tutorials/calico-policy-advanced.md

```diff
@@ -0,0 +1,345 @@
+
+### Requirements
+
+- Calico v2.6.1+ with Kubernetes 1.8+
+- A working Kubernetes cluster and access to it using kubectl
+- Your Kubernetes nodes have connectivity to the public internet
+- You are familiar with [Kubernetes NetworkPolicy](kubernetes-policy-basic) or [Kubernetes NetworkPolicy Advanced](kubernetes-policy-advanced)
+
+### Tutorial flow
+
+1. Create the Namespace and Nginx Service
+1. Deny all ingress traffic
+1. Allow ingress traffic to Nginx
+1. Deny all egress traffic
+1. Allow egress traffic to kube-dns
+1. Cleanup Namespace
+
+### 1. Create the namespace and nginx service
+
+We'll use a new namespace for this guide. Run the following commands to create it and a plain nginx service listening on port 80.
+
+```bash
+kubectl create ns advanced-policy-demo
+kubectl run --namespace=advanced-policy-demo nginx --replicas=2 --image=nginx
+kubectl expose --namespace=advanced-policy-demo deployment nginx --port=80
+```
+
+#### Verify access - allowed all ingress and egress
+
+Open up a second shell session which has `kubectl` connectivity to the Kubernetes cluster and create a busybox pod to test policy access. This pod will be used throughout this tutorial to test policy access.
+
+```bash
+kubectl run --namespace=advanced-policy-demo access --rm -ti --image busybox /bin/sh
+```
+
+This should open up a shell session inside the `access` pod, as shown below.
+
+```
+Waiting for pod advanced-policy-demo/access-472357175-y0m47 to be running, status is Pending, pod ready: false
+
+If you don't see a command prompt, try pressing enter.
+/ #
+```
+
+{: .no-select-button}
+
+Now from within the busybox "access" pod execute the following command to test access to the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It should return the HTML of the nginx welcome page.
+
+Still within the busybox "access" pod, issue the following command to test access to google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It should return the HTML of the google.com home page.
+
+### 2. Deny all ingress traffic
+
+Enable ingress isolation on the namespace by deploying a [default deny all ingress traffic policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/#default-deny-all-ingress-traffic).
+
+```bash
+kubectl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: default-deny-ingress
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  ingress:
+  - action: Deny
+EOF
+```
+
+#### Verify access - denied all ingress and allowed all egress
+
+Because all pods in the namespace are now selected, any ingress traffic which is not explicitly allowed by a policy will be denied.
+
+We can see that this is the case by switching over to our "access" pod in the namespace and attempting to access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It should return:
+
+```bash
```

Comment:
nit: could you remove the `bash` bit from this line? This is to be consistent with the way command output is rendered elsewhere on this page (i.e. without the prompt).

Reply:
Done! Thank you very much for the review